src/GafferOSL/ShadingEngine.cpp

```diff
@@ -1241,6 +1508,222 @@ IECore::CompoundDataPtr ShadingEngine::shade( const IECore::CompoundData *points
 	return results.results();
 }
 
+template< int WidthT >
+IECore::CompoundDataPtr executeShadeBatched( const IECore::CompoundData *points, const ShadingEngine::Transforms &transforms, ShaderGroup &shaderGroup, ShadingSystem *shadingSystem, const std::vector<IECore::InternedString> &contextVariablesNeeded )
```

Comment:
It looks like we could replace the `ShadingSystem` argument with `BatchedExecutor<WidthT>` - that seems a bit smpler to me, and makes it clear that we're not actually using the shading system directly.